app/models/usable_slot.rb

```diff
@@ -0,0 +1,73 @@
+class UsableSlot < ApplicationRecord
+  belongs_to :user
+
+  def self.exact_match(start_at, end_at)
+    where(start_at: start_at, end_at: end_at)
+  end
+
+  def self.regenerate_for_six_weeks
+    User.guiders.each do |user|
+      generate_for_guider(user)
+    end
+  end
+
+  def self.within_date_range(from, to)
+    where('usable_slots.start_at > ? AND usable_slots.end_at < ?', from, to)
+  end
+
+  def self.usable
+    where(appointment_id: nil)
+  end
+
+  def self.without_appointments
+    UsableSlot
+      .joins(<<-SQL
+              LEFT JOIN appointments ON
+                appointments.user_id = usable_slots.user_id AND
+                appointments.start_at = usable_slots.start_at AND
+                appointments.end_at = usable_slots.end_at
+              SQL
+            )
+      .where('appointments.start_at IS NULL')
+  end
+
+  def self.with_guider_count(from, to)
+    UsableSlot
+      .select('DISTINCT usable_slots.start_at, usable_slots.end_at, count(1) AS guiders')
+      .without_appointments
+      .group('usable_slots.start_at, usable_slots.end_at')
+      .within_date_range(from, to)
+      .map do |us|
+      { guiders: us.attributes['guiders'], start: us.start_at, end: us.end_at }
+    end
+  end
+
+  private_class_method
```

Comment:
This has no effect unless you pass the method names as symbols. It's not actually a language keyword as `private` is, just a regular method.